R/data_tabulate.R

```diff
@@ -465,6 +483,87 @@ as.data.frame.datawizard_tables <- function(x,
 as.data.frame.datawizard_crosstabs <- as.data.frame.datawizard_tables
 
 
+# as.table --------------------
+
+#' @rdname data_tabulate
+#' @export
+as.table.datawizard_table <- function(x, remove_na = TRUE, ...) {
+  # sanity check - the data frame method returns a list
+  if (!is.data.frame(x)) {
+    x <- x[[1]]
+  }
+  if (remove_na) {
+    # remove NA values from the table
+    x <- x[!is.na(x$Value), ]
+  }
+  as.table(stats::setNames(x[["N"]], x$Value))
+}
+
+#' @export
+as.table.datawizard_tables <- function(x, remove_na = TRUE, ...) {
+  out <- lapply(x, as.table.datawizard_table, remove_na = remove_na, ...)
+  # if only one table is returned, "unlist"
+  if (length(out) == 1) {
+    out <- out[[1]]
+  }
```

Comment:
Since the user passes a list as input, it seems weird to me to return something else than a list. Is it a problem to let the user handle the case of length 1 output?

Reply:
I think `as.table()` should return a table, when the input also is only "one frequency table", and if the user computed *multiple* frequency tables, then `as.table()` should return a list of tables. The user usually doesn't know that the default-method of `data_tabulate()` returns a data frame, while the `.data.frame` method returns a list with one data frame (or more data frames for multiple tables).